src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,46 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private final K [] keys;
+    private final V [] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_ITEMS_NUMBER];
+        values = (V[]) new Object[MAX_ITEMS_NUMBER];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (get(key) == null) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        }
+        for (int i = 0; i < keys.length; i++) {
+            if (Objects.equals(keys[i], key)) {
+                keys[i] = key;
+                values[i] = value;
+            }
+        }
```

Comment:
Немає необхідності двічі проходитися по циклу
```suggestion
        for (int i = 0; i < keys.length; i++) {
            if (Objects.equals(keys[i], key)) {
                values[i] = value;
                return;
            }
        }
        keys[size] = key;
        values[size] = value;
        size++;
```

Reply:
Так не проходит тест